csv.go

```diff
@@ -0,0 +1,180 @@
+package main
+
+import (
+	"encoding/csv"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+)
+
+type csvPrinter struct {
+	writer     *csv.Writer
+	file       *os.File
+	filename   string
+	headerDone bool
+}
+
+const (
+	csvTimeFormat = "2006-01-02 15:04:05.000"
+)
+
+func newCSVPrinter(filename string, args []string) (*csvPrinter, error) {
+	file, err := os.Create(filename)
+	if err != nil {
+		return nil, fmt.Errorf("error creating CSV file: %w", err)
+	}
+
+	writer := csv.NewWriter(file)
+	return &csvPrinter{
+		writer:     writer,
+		file:       file,
+		filename:   filename,
+		headerDone: false,
+	}, nil
+}
+
+func (cp *csvPrinter) writeHeader() error {
+	header := []string{
+		"Event Type", "Timestamp", "Address", "Hostname", "Port",
+		"Hostname Resolve Retries", "Total Successful Probes", "Total Unsuccessful Probes",
+		"Never Succeed Probe", "Never Failed Probe", "Last Successful Probe",
+		"Last Unsuccessful Probe", "Total Packets", "Total Packet Loss",
+		"Total Uptime", "Total Downtime", "Longest Uptime", "Longest Uptime Start",
+		"Longest Uptime End", "Longest Downtime", "Longest Downtime Start",
+		"Longest Downtime End", "Latency Min", "Latency Avg", "Latency Max",
+		"Start Time", "End Time", "Total Duration",
+	}
+	return cp.writer.Write(header)
+}
+
+func (cp *csvPrinter) saveStats(tcping tcping) error {
+	if !cp.headerDone {
+		if err := cp.writeHeader(); err != nil {
+			return fmt.Errorf("error writing CSV header: %w", err)
+		}
+		cp.headerDone = true
+	}
+
+	totalPackets := tcping.totalSuccessfulProbes + tcping.totalUnsuccessfulProbes
+	packetLoss := float64(tcping.totalUnsuccessfulProbes) / float64(totalPackets) * 100
```

Comment:
```suggestion
	packetLoss := float32(tcping.totalUnsuccessfulProbes) / float32(totalPackets) * 100
```

To make it uniform with the rest of the code base